### PR TITLE
[DOCS-3324] Document agent dual shipping features

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?

Document the dual shipping features present in the agent.

### Motivation

Some users want to send data to multiple places, the agent can do this but we need to document it. 

### Preview

https://docs-staging.datadoghq.com/brian.floersch/dual-shipping/agent/guide/dual-shipping/?tab=metricsservicechecks

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by olivielpeau** [comment]
Todo: mention explicitly that with the helm chart the yaml section can be used as a configmap

File: content/en/agent/guide/dual-shipping.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+<div class="alert alert-danger">
+Dual shipping can impact billing if you are sending data to multiple Datadog orgs. For more information about the impact of this configuration, contact <a href="/help/">Datadog Support</a>.
+</div>
+
+## Overview
+
+If you wish to send data to more than one destination such as a second Datadog org or other internal infrastructure - you can configure the Agent to send data to additional endpoints. Most data sent from the Agent supports multiple endpoints or API keys.
+
+
+## Agent Metrics
```
Comment (olivielpeau) [inline content/en/agent/guide/dual-shipping.md:15]:
```suggestion
## Metrics/APM/Live Processes/Orchestrator
```

File: content/en/agent/guide/dual-shipping.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+```bash
+DD_APM_ADDITIONAL_ENDPOINTS='{\"https://mydomain.datadoghq.com\": [\"apikey2\", \"apikey3\"], \"https://mydomain.datadoghq.eu\": [\"apikey4\"]}'
+
+DD_APM_PROFILING_ADDITIONAL_ENDPOINTS='{\"https://mydomain.datadoghq.com\": [\"apikey2\", \"apikey3\"], \"https://mydomain.datadoghq.eu\": [\"apikey4\"]}'
+```
+
+{{% /tab %}}
+
+<!-- Process Agent -->
+{{% tab "Process Agent" %}}
```
Comment (olivielpeau) [inline content/en/agent/guide/dual-shipping.md:76]:
```suggestion
{{% tab "Live Processes" %}}
```

File: content/en/agent/guide/dual-shipping.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+```bash
+DD_NETWORK_DEVICES_METADATA_USE_HTTP=true
+DD_NETWORK_DEVICES_METADATA_ADDITIONAL_ENDPOINTS="[{\"api_key\": \"apiKey2\", \"Host\": \"https://mydomain.datadoghq.com\", \"Port\": 443, \"is_reliable\": true}]"
+```
+
+{{% /tab %}}
+
+<!-- Compliance -->
+{{% tab "Compliance" %}}
```
Comment (olivielpeau) [inline content/en/agent/guide/dual-shipping.md:222]:
```suggestion
{{% tab "CSPM" %}}
```

File: content/en/agent/guide/dual-shipping.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+```bash
+DD_​​COMPLIANCE_CONFIG_ENDPOINTS_USE_HTTP=true
+DD_​​COMPLIANCE_CONFIG_ENDPOINTS_ADDITIONAL_ENDPOINTS="[{\"api_key\": \"apiKey2\", \"Host\": \"https://mydomain.datadoghq.com\", \"Port\": 443, \"is_reliable\": true}]"
+```
+
+{{% /tab %}}
+
+<!-- Runtime Security -->
+{{% tab "Runtime Security" %}}
```
Comment (olivielpeau) [inline content/en/agent/guide/dual-shipping.md:247]:
```suggestion
{{% tab "CWS" %}}
```

File: content/en/agent/guide/dual-shipping.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+```bash
+DD_ORCHESTRATOR_EXPLORER_ORCHESTRATOR_ADDITIONAL_ENDPOINTS='{\"https://mydomain.datadoghq.com\": [\"apikey2\", \"apikey3\"], \"https://mydomain.datadoghq.eu\": [\"apikey4\"]}'
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+You can add the yaml configuration to your `datadog.yaml` or launch the agent with the appropriate environment variables. 
+## Logs and Events
```
Comment (olivielpeau) [inline content/en/agent/guide/dual-shipping.md:125]: Logs/DBM/[...]

File: content/en/agent/guide/dual-shipping.md [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+
+## Logs, Database Monitoring, Network Devices, CSPM, Runtime Security
+
+For data from these products, when setting up additional endpoints, you must explicitly set `use_http` to tell the Agent which transport to use. The same transport configuration is shared among all additional endpoints.
+
+The `is_reliable` setting tells the Agent to treat this endpoint with the same priority as the primary endpoint. The primary endpoint is always reliable. The `is_reliable` setting ensures that logs are not missed if a destination becomes unavailable. 
+
+For example, if you're sending logs to two reliable endpoints and one becomes unavailable, logs continue to flow to the second reliable endpoint. If both endpoints become unavailable, the logs stop flowing until at least one endpoint recovers.
+
+The `is_reliable` setting defaults to `false`. Unreliable endpoints only send logs if they successfully send to at least one reliable endpoint. You may define multiple additional endpoints with mixed use of `is_reliable`.
```
Comment (hestonhoffman) [inline content/en/agent/guide/dual-shipping.md:129]: Could you clarify this, or rephrase? It's not super clear.

**Review by hestonhoffman — CHANGES_REQUESTED** [review]
I pushed up a commit with a bunch of minor edits. Just one question for you.

File: content/en/agent/guide/dual-shipping.md (hunk 6 above)
Comment (gh123man) [reply content/en/agent/guide/dual-shipping.md:129]: I tried to clarify as best I could. It's a complicated behavior so it's difficult to summarize in simple terms. Let me know if this is better.

**Review by hestonhoffman — APPROVED** [review]

**Comment by gh123man** [comment]
@hestonhoffman Great thanks! Yeah I think everything looks good. Going to hold this util tomorrow (Ive reached out to a few others for comment) but if I don't get any more feedback by tomorrow, ill remove the `Do Not Merge` tag. Thanks for the review and edits!

File: content/en/agent/guide/dual-shipping.md [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+## Logs, Database Monitoring, Network Devices, CSPM, Runtime Security
+
+For data from these products, when setting up additional endpoints, you must explicitly set `use_http` to tell the Agent which transport to use. The same transport configuration is shared among all additional endpoints.
+
+The `is_reliable` setting tells the Agent to treat this endpoint with the same priority as the primary endpoint. The primary endpoint is always reliable. The `is_reliable` setting ensures that logs are not missed if a destination becomes unavailable. 
+
+For example, if you're sending logs to the main endpoint and an additional endpoint with `is_reliable: true` and one endpoint becomes unavailable, logs continue to flow to the other endpoint. If both endpoints become unavailable, the Agent stops reading and sending logs until at least one endpoint recovers. This ensures all logs make it to at least one reliable endpoint.
+
+The `is_reliable` setting defaults to `false`. Unreliable endpoints only send logs if at least one reliable endpoint is available. You may define multiple additional endpoints with mixed use of `is_reliable`.
+
```
Comment (olivielpeau) [inline content/en/agent/guide/dual-shipping.md:130]: For clarity I think we should repeat here:

```suggestion

You can add the YAML configuration to your `datadog.yaml` or launch the Agent with the appropriate environment variables.

```

File: content/en/agent/guide/dual-shipping.md [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+## Logs, Database Monitoring, Network Devices, CSPM, Runtime Security
+
+For data from these products, when setting up additional endpoints, you must explicitly set `use_http` to tell the Agent which transport to use. The same transport configuration is shared among all additional endpoints.
+
+The `is_reliable` setting tells the Agent to treat this endpoint with the same priority as the primary endpoint. The primary endpoint is always reliable. The `is_reliable` setting ensures that logs are not missed if a destination becomes unavailable. 
+
+For example, if you're sending logs to the main endpoint and an additional endpoint with `is_reliable: true` and one endpoint becomes unavailable, logs continue to flow to the other endpoint. If both endpoints become unavailable, the Agent stops reading and sending logs until at least one endpoint recovers. This ensures all logs make it to at least one reliable endpoint.
+
+The `is_reliable` setting defaults to `false`. Unreliable endpoints only send logs if at least one reliable endpoint is available. You may define multiple additional endpoints with mixed use of `is_reliable`.
```
Comment (olivielpeau) [inline content/en/agent/guide/dual-shipping.md:129]: I'd add a recommendation to use `is_reliable: true`:
```suggestion
The `is_reliable` setting defaults to `false`, and is recommended to be set to `true`. Unreliable endpoints only send logs if at least one reliable endpoint is available. You may define multiple additional endpoints with mixed use of `is_reliable`.
```

**Review by olivielpeau — APPROVED** [review]
Left a few comments, but they can all be addressed later in a separate PR, the current PR LGTM! Thanks!

File: content/en/agent/guide/dual-shipping.md [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+```
+
+{{% /tab %}}
+{{% /tabs %}} 
+
+## Logs, Database Monitoring, Network Devices, CSPM, Runtime Security
+
+For data from these products, when setting up additional endpoints, you must explicitly set `use_http` to tell the Agent which transport to use. The same transport configuration is shared among all additional endpoints.
+
+The `is_reliable` setting tells the Agent to treat this endpoint with the same priority as the primary endpoint. The primary endpoint is always reliable. The `is_reliable` setting ensures that logs are not missed if a destination becomes unavailable.
```
[the remark this answers is not in the document]
Comment (gh123man) [reply content/en/agent/guide/dual-shipping.md:125]: That's how I originally formatted it - [but it was moved in the editing pass](https://github.com/DataDog/documentation/pull/13371/commits/37c0cdc336a6a5bf2fba9a6aabdcb3a9fe129593). 

@hestonhoffman what do you think?

[the remark this answers is not in the document]
Comment (hestonhoffman) [reply content/en/agent/guide/dual-shipping.md:125]: @gh123man Yep, no problem. We can move it back to below the config

File: content/en/agent/guide/dual-shipping.md [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+DD_​​RUNTIME_SECURITY_CONFIG_ENDPOINTS_USE_HTTP=true
+DD_​​RUNTIME_SECURITY_CONFIG_ENDPOINTS_ADDITIONAL_ENDPOINTS="[{\"api_key\": \"apiKey2\", \"Host\": \"https://mydomain.datadoghq.com\", \"Port\": 443, \"is_reliable\": true}]"
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+For data from these products, when setting up additional endpoints, you must explicitly set `use_http` to tell the Agent which transport to use. The same transport configuration is shared among all additional endpoints.
+
+The `is_reliable` setting tells the Agent to treat this endpoint with the same priority as the primary endpoint. The primary endpoint is always reliable. The `is_reliable` setting ensures that logs are not missed if a destination becomes unavailable.
```
Comment (hestonhoffman) [inline content/en/agent/guide/dual-shipping.md:266]:
```suggestion
The `is_reliable` setting tells the Agent to treat this endpoint with the same priority as the primary endpoint. The primary endpoint is always reliable. This ensures that logs are not missed if a destination becomes unavailable. 
```

File: content/en/agent/guide/dual-shipping.md [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+{{% /tabs %}}
+
+For data from these products, when setting up additional endpoints, you must explicitly set `use_http` to tell the Agent which transport to use. The same transport configuration is shared among all additional endpoints.
+
+The `is_reliable` setting tells the Agent to treat this endpoint with the same priority as the primary endpoint. The primary endpoint is always reliable. The `is_reliable` setting ensures that logs are not missed if a destination becomes unavailable. 
+
+For example, if you're sending logs to the main endpoint and an additional endpoint with `is_reliable: true` and one endpoint becomes unavailable, logs continue to flow to the other endpoint. If both endpoints become unavailable, the Agent stops reading and sending logs until at least one endpoint recovers. This ensures all logs make it to at least one reliable endpoint.
+
+The `is_reliable` setting defaults to `false`, and is recommended to be set to `true`. Unreliable endpoints only send logs if at least one reliable endpoint is available. You may define multiple additional endpoints with mixed use of `is_reliable`.
+
```
Comment (hestonhoffman) [inline content/en/agent/guide/dual-shipping.md:271]:
```suggestion
The `is_reliable` setting defaults to `false`. We recommended you set `is_reliable` to `true`. Unreliable endpoints only send logs if at least one reliable endpoint is available. You may define multiple additional endpoints with mixed use of `is_reliable`.

```

**Review by hestonhoffman — APPROVED** [review]
A couple of small suggestions, but this is GTG

**Pull request merged**